cfn-resources/stream-instance/cmd/resource/resource.go

```diff
@@ -0,0 +1,318 @@
+		Name:    model.Name,
+		GroupId: model.GroupId,
+		DataProcessRegion: &admin.StreamsDataProcessRegion{
+			CloudProvider: *dataProcessRegion.CloudProvider,
+			Region:        *dataProcessRegion.Region,
+		},
+		StreamConfig: &admin.StreamConfig{
+			Tier: streamConfig.Tier,
+		},
+	}
+}
+
+func newModelDataRegion(dataProcessRegion *admin.StreamsDataProcessRegion) *StreamsDataProcessRegion {
+	return &StreamsDataProcessRegion{
+		CloudProvider: &dataProcessRegion.CloudProvider,
+		Region:        &dataProcessRegion.Region,
+	}
+}
+
+func newModelStreamConfig(streamConfig *admin.StreamConfig) *StreamConfig {
+	return &StreamConfig{
+		Tier: streamConfig.Tier,
+	}
+}
+
+func newModelDBRoleToExecute(dbRole *admin.DBRoleToExecute) *DBRoleToExecute {
+	return &DBRoleToExecute{
+		Role: dbRole.Role,
+		Type: dbRole.Type,
+	}
+}
+
+func newModelAuthentication(authentication *admin.StreamsKafkaAuthentication) *StreamsKafkaAuthentication {
+	return &StreamsKafkaAuthentication{
+		Mechanism: authentication.Mechanism,
+		Password:  authentication.Password,
+		Username:  authentication.Username,
+	}
+}
+
+func newModelSecurity(security *admin.StreamsKafkaSecurity) *StreamsKafkaSecurity {
+	return &StreamsKafkaSecurity{
+		BrokerPublicCertificate: security.BrokerPublicCertificate,
+		Protocol:                security.Protocol,
+	}
+}
+
+func newModelConnections(streamConfig *[]admin.StreamsConnection) []StreamsConnection {
+	if streamConfig == nil || len(*streamConfig) == 0 {
+		return nil
+	}
+
+	connections := make([]StreamsConnection, 0)
+	for _, connection := range *streamConfig {
+		modelConnection := StreamsConnection{
+			Name: connection.Name,
+			Type: connection.Type,
+		}
+		if connection.GetType() == Cluster {
+			modelConnection.ClusterName = connection.ClusterName
+			modelConnection.DbRoleToExecute = newModelDBRoleToExecute(connection.DbRoleToExecute)
+		} else if connection.GetType() == Kafka {
+			modelConnection.BootstrapServers = connection.BootstrapServers
+			modelConnection.Authentication = newModelAuthentication(connection.Authentication)
+			modelConnection.Security = newModelSecurity(connection.Security)
+		}
+		connections = append(connections, modelConnection)
+	}
+	return connections
+}
+
+func handleError(response *http.Response, method constants.CfnFunctions, err error) (handler.ProgressEvent, error) {
+	errMsg := fmt.Sprintf("%s error:%s", method, err.Error())
+	_, _ = logger.Warn(errMsg)
+	if response.StatusCode == http.StatusConflict {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          errMsg,
+			HandlerErrorCode: cloudformation.HandlerErrorCodeAlreadyExists}, nil
+	}
+	if response.StatusCode == http.StatusBadRequest {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          errMsg,
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+	}
+	return progressevent.GetFailedEventByResponse(errMsg, response), nil
```

Comment:
Good point. Reviewing it I removed the custom handling of bad request. But keeping `StatusConflict` because `StatusConflict` is not mapped to any specific error code like in this method and defaults to `HandlerErrorCodeInternalFailure`

Reply:
Probably a good future improvement and opportunity to clean up some code to include in `GetFailedEventByResponse` all "custom" error handling done in different resources

Reply:
agree, I ended up having to handle some custom error handling for my resource as well so would be good to check if we have room for further generalization.